src/samples/network/echo-client.c

```diff
@@ -50,16 +50,16 @@ static bool
 on_can_read(void *data, struct sol_socket *s)
 {
     ssize_t r;
-    char buffer[1024] = { };
+    struct sol_buffer buffer = SOL_BUFFER_INIT_EMPTY;
     struct sol_network_link_addr addr;
 
-    r = sol_socket_recvmsg(s, buffer, 1024, &addr);
+    r = sol_socket_recvmsg(s, &buffer, &addr);
     if (r < 0) {
         fprintf(stderr, "ERROR: Failed in receiving the message\n");
         goto err;
     }
 
-    printf("Received: %s\n", buffer);
+    printf("Received: %.*s\n", (int)buffer.used, (char *)buffer.data);
```

Comment:
it's longer, but nothing prevents you from adding a macro that does SOL_BUFFER_PRINT()